cmake/MachineCfg.cmake

```diff
@@ -21,11 +26,15 @@ elseif (DEFINED ENV{MACHINE_CFG})
 endif()
 
 if (USE_MACHINE_CFG)
+  # Get the Parthenon root directory, which is one directory up from
+  # from the "cmake" directory. This is done in-case MachineCfg.cmake is
+  # included from a project using Parthenon.
+  get_filename_component(PAR_ROOT ${CMAKE_CURRENT_LIST_DIR} DIRECTORY)
+
   # The `@PAR_ROOT@` string allows the devleper to specify a relative path to
-  # the project directory via an environment variable.
+  # the parthenon root directory via an environment variable.
   string(
-    REPLACE "@PAR_ROOT@" ${CMAKE_CURRENT_SOURCE_DIR}
-      USE_MACHINE_CFG ${USE_MACHINE_CFG})
+    REPLACE "@PAR_ROOT@" ${PAR_ROOT} USE_MACHINE_CFG ${USE_MACHINE_CFG})
```

Comment:
Out of curiosity, what is actually happening here? For what kind of scenario do we need this line?

If I understand correctly 
- USE_MACHINE_CFG could be sth like `"../myfile/machine.cmake"`
- `${PAR_ROOT}` now always contains the `/path/to/parthenon`  because from this file (`MachineCfg.cmake`) is included from the `/path/to/parthenon/CMakeLists.txt` and the 
- `@PAR_ROOT@` is replaced by whatever `PAR_ROOT` is set to by some external project that uses parthenon.
  - so if `@PAR_ROOT@` has the full path, it does already need to match `${PAR_ROOT}` by construction (otherwise the file that included this `MachineCfg.cmake` would not be the Parthenon `CmakeLists.txt` but some other, which is likely not intended behavior)
  - if `@PAR_ROOT@` contains a relative path then it'd replace the relative path of `USE_MACHINE_CFG` with the full path only if a custom `machine.cmake` was used in the Parthnenon subdirectory

We get the full path below anyway, so I'm just curious if this line is still required.

Reply:
ping @AndrewGaspar (I just went through the backlog and except for this comment this looks good)